tfhe/src/core_crypto/commons/math/random/tests.rs

```diff
@@ -504,17 +547,152 @@ fn test_uniform_random_custom_mod<
+        let distinct_values: usize = if ciphertext_modulus.is_native_modulus() {
+            assert!(
+                Scalar::BITS <= usize::BITS as usize,
+                "Unable to run test for such a large modulus {ciphertext_modulus:?}, usize::MAX {}",
+                usize::MAX
+            );
+            1 << Scalar::BITS
+        } else {
+            ciphertext_modulus.get_custom_modulus().cast_into()
+        };
+
+        distinct_values
+    }
+
+    fn cumulative_distribution_function(
+        &self,
+        integer_value: Scalar,
+        distinct_values: usize,
+        ciphertext_modulus: CiphertextModulus<Scalar>,
+    ) -> f64 {
+        // This is only valid if the smallest integer value for uniform is 0 which is currently the
+        // case
+        let value_as_usize = self.map_value_to_usize(integer_value, ciphertext_modulus);
+        let integer_f64 = (value_as_usize + 1) as f64;
+        let distinct_values_f64: f64 = distinct_values as f64;
+
+        integer_f64 / distinct_values_f64
+    }
+
+    fn map_usize_to_value(
+        &self,
+        input: usize,
+        _ciphertext_modulus: CiphertextModulus<Scalar>,
+    ) -> Scalar {
+        Scalar::cast_from(input)
+    }
+
+    fn map_value_to_usize(
+        &self,
+        input: Scalar,
+        _ciphertext_modulus: CiphertextModulus<Scalar>,
+    ) -> usize {
+        input.cast_into()
+    }
+}
+
 // This test takes care of all bigger native types, as underneath the CSPRNG outputs bytes
 #[test]
 fn test_uniform_random_native_mod_u8() {
-    test_uniform_random_custom_mod::<u8>(CiphertextModulus::new_native());
+    let ciphertext_modulus = CiphertextModulus::new_native();
+    test_random_from_distribution_custom_mod::<u8, Uniform>((), ciphertext_modulus);
 }
 
 #[test]
 fn test_uniform_random_custom_mod_u64() {
     // Have a modulus that will generate some rejection but is relatively small to be able to test
     // quickly
-    test_uniform_random_custom_mod::<u64>(
-        CiphertextModulus::try_new((1 << 10) + (1 << 9)).unwrap(),
-    );
+    let ciphertext_modulus = CiphertextModulus::try_new((1 << 10) + (1 << 9)).unwrap();
+    test_random_from_distribution_custom_mod::<u64, Uniform>((), ciphertext_modulus);
+}
+
+impl<Scalar: UnsignedInteger + CastFrom<usize> + CastInto<usize>> DistributionTestHelper<Scalar>
+    for TUniform<Scalar>
+{
+    type CreationInfos = u32;
+
+    fn new_with_custom_modulus(
+        value: Self::CreationInfos,
+        ciphertext_modulus: CiphertextModulus<Scalar>,
+    ) -> Self {
+        assert!(ciphertext_modulus.is_native_modulus());
+        Self::new(value)
+    }
+
+    fn distinct_values(&self, _ciphertext_modulus: CiphertextModulus<Scalar>) -> usize {
+        // TUniform does not support non native moduli for now
+        self.distinct_value_count()
+    }
+
+    fn cumulative_distribution_function(
+        &self,
+        integer_value: Scalar,
+        _distinct_values: usize,
```

Comment:
Shouldn't `distinct_values` be removed from this function signature in the trait ?

Reply:
yeah I guess since it's a method on the trait now, it may not have been before, thanks for catching that !

Reply:
fixed